subprojects/docs/src/docs/userguide/upgrading_version_5.adoc

```diff
@@ -59,6 +59,14 @@ Its predecessor `IncrementalTaskInputs` has been deprecated.
 Forcing dependency versions using `force = true` on a first-level dependency is deprecated.
 Force has both a semantic and ordering issue which can be avoided by using a <<rich_versions.adoc#rich-version-constraints, strict version constraint>>.
 
+==== Invalid task definitions and configurations
+
+Problems with task definitions are called out in deprecation warnings like this:
```

Comment:
More context would be nice. And a link to the user manual how to fix these problems. I think the runtime validation is a feature as well, so it should go to the release notes and we should have some documentation in the user manual. This can be taken care of in a follow up PR.